### ISSUE TITLE
cordova prepare --nofetch not working since 2.3.5 update

### ISSUE THREAD

**Issue by sbatezat** [mapsplugin/cordova-plugin-googlemaps#2359]
**I'm submitting a ...**  (check one with "x")
- [ ] question
- [ X] any problem or bug report

**cordova information:**  (run `$> cordova plugin list`) 

```
cordova-android-support-gradle-release 1.4.4 "cordova-android-support-gradle-release"
cordova-plugin-app-event 1.2.1 "Application Events"
cordova-plugin-app-version 0.1.9 "AppVersion"
cordova-plugin-camera 2.4.1 "Camera"
cordova-plugin-compat 1.2.0 "Compat"
cordova-plugin-crosswalk-webview 2.3.0 "Crosswalk WebView Engine"
cordova-plugin-device 1.1.5 "Device"
cordova-plugin-file 4.3.2 "File"
cordova-plugin-google-analytics 1.8.3 "Google Universal Analytics Plugin"
cordova-plugin-googlemaps 2.3.8 "cordova-plugin-googlemaps"
cordova-plugin-ignore-lint-translation 0.0.1 "cordova-plugin-ignore-lint-translation"
cordova-plugin-image-picker 1.1.1 "ImagePicker"
cordova-plugin-insomnia 4.3.0 "Insomnia (prevent screen sleep)"
cordova-plugin-nativestorage 2.3.2 "NativeStorage"
cordova-plugin-network-information 1.3.3 "Network Information"
cordova-plugin-screen-orientation 2.0.2-dev "Screen Orientation"
cordova-plugin-splashscreen 4.0.3 "Splashscreen"
cordova-plugin-whitelist 1.3.2 "Whitelist"
cordova-plugin-x-socialsharing 5.1.8 "SocialSharing"
de.appplant.cordova.plugin.local-notification 0.8.5 "LocalNotification"
es6-promise-plugin 4.2.2 "Promise"
ionic-plugin-keyboard 2.2.1 "Keyboard"
phonegap-plugin-mobile-accessibility 1.0.4 "Mobile Accessibility"
uk.co.workingedge.phonegap.plugin.istablet 1.2.0 "Is Tablet"
```

**Current behavior:**

Remove your platforms and plugins folder, and run the following cordova command to restore them :
cordova prepare --nofetch is throwing an error :
```
Failed to install 'cordova-plugin-googlemaps': Error: Command failed: npm install xml2js minimist --save 2>&1

It doesn't really matter but please note the error is thrown even if xml2js & minimist are already defined on the package.json file and installed (npm i).
```

**Expected behavior:**
No error.

I'm using cordova 7.0.1.

**Comment by wf9a5m75** [comment]
Problem is your cordova version and the maps plugin versions are too old.
You can remove the maps plugin with like this:

```
$> cordova platform rm android ios

$> cordova plugin rm cordova-plugin-googlemaps

$> cordova plugin add cordova-plugin-googlemaps@2.3.8 --variable ...

$> cordova platform add android ios
```

**Comment by sbatezat** [comment]
Thanks for your quick answer but :
- See my cordova plugin list, I'm using the latest plugin version.
- I'm not sure my app and my build process will be fine if I'm doing the upgrade to the latest cordova version. I'll double check and keep you in touch. However, I understand it's not on you if it's really the source of the issue, but maybe adding the cordova version required on the readme will prevent any new issue about that?

**Comment by wf9a5m75** [comment]
I think the problem is in your environment. **"IF"** this error occurs for all people or many people, I should receive more the same reports.

**Comment by wf9a5m75** [comment]
And you know what, `npm 4.2.0` is too old. The current npm is `6.0.1`.

**Comment by sbatezat** [comment]
Well, maybe it's my environment, but 2.3.5 (first revision of your plugin that is breaking cordova prepare command on my environment) is not that old and most of people never use cordova prepare command so maybe not.

I know npm 4.2.0 is old. Node 7.10.1 is old too, cordova 7.1.0 is old too, etc...
The thing is dependencies for npm X are not always compatible with npm X+1. So when I get a legacy project, I deal with its old dependencies version except very good reason to not (because I don't have days to fix issues related to any migration).

If you don't support node, npm, cordova, etc bellow a version number, I think it would be a good idea to add this on the readme. From my point of view, a cordova plugin should only rely on cordova packages (cordova, cordova-android, cordova-ios in your case) and not on npm & node installations (https://en.wikipedia.org/wiki/Separation_of_concerns) 

Also please note this is working on my enviroment: https://github.com/sbatezat/cordova-plugin-googlemaps/commit/8b57760e352064bd2dc3ae30aee57cce08e87fa1. It's not really a fix because I had added dependencies manually, but well, it's working for now like that.

**Comment by wf9a5m75** [comment]
`cordova prepare` command is executed automatically before `cordova run (platform)` or `cordova build (platform)` , so everyone execute the command actually.

**Comment by sbatezat** [comment]
Yeah I know, what I mean is that most of people are not using that command to restore platforms & plugins. When cordova prepare is executed, if plugins and platforms are already added, there is no issue

**Comment by wf9a5m75** [comment]
Since I can't test all npm versions, all node versions, all environments, all devices.. (people ask me to do that without money), it's impossible. That's why all source code is full open.
Please fork and modify the code for your environment.

So far I can't accept your claim, but I keep in mind your report.
